Sources/SpeziAccount/AccountSetup.swift

```diff
@@ -61,34 +71,50 @@ public struct AccountSetup<Header: View, Continue: View>: View {
     }
 
     public var body: some View {
-        NavigationStack {
-            GeometryReader { proxy in
-                ScrollView(.vertical) {
-                    VStack {
-                        if !services.isEmpty {
-                            header
-                        }
+        GeometryReader { proxy in
+            ScrollView(.vertical) {
+                VStack {
+                    if !services.isEmpty {
+                        header
+                            .environment(\._accountSetupState, setupState)
+                    }
 
-                        Spacer()
+                    Spacer()
 
-                        if let details = account.details {
+                    if let details = account.details {
+                        if case .loadingExistingAccount = setupState {
+                            // We allow the outer view to navigate away upon signup, before we show the existing account view
+                            ProgressView()
+                                .task {
+                                    try? await Task.sleep(for: .seconds(2))
```

Comment:
Do you know what the rough timing estimate for a navigation animation might be there? I am fine with 2 seconds but it might be a bit too long if the animation itself is way shorter.

Reply:
Just testing with the template application I found, that navigation pretty much happens instantly. So SwiftUI doesn't wait for the sheet to disappear and immediately jumps to the next onboarding step.
In the TestApps for example, we render the AccountSetup in a sheet as well. In this case, SwiftUI first dismisses the SignupForm completely and then dismisses the AccountSetup view. Each animation happens in sequence.
So animations (depending on the situation) should roughly be finished in about a second. I think setting it to 2 doesn't hurt and accommodates for any lag on e.g. older devices.

So I assume I can mark this conversation as resolved and merge once the tests finish?

Reply:
Sounds good, thank you for the context @Supereg!